[BottomNavigation] Remove `wrapper` from BottomNavigationAction (#26923)
mui-org/material-ui: docs/src/pages/guides/migration-v4/migration-v4.md

  +<BottomNavigation onChange={(event: React.SyntheticEvent) => {}} />
  ```

> - Remove the `span` element that wraps the children. Remove the `wrapper` classKey too. More details about [this change](https://github.com/mui-org/material-ui/pull/26666).

Reviewer: I think it's confusing linking different PR here. Let's update the current PR descirption with the details of why the changes was introduced and link this PR in the migration guide.

  +<BottomNavigation onChange={(event: React.SyntheticEvent) => {}} />
  ```

> ### BottomNavigationAction
> 
> - `span` element that wraps children has been removed. `wrapper` classKey is also removed. More details about [this change](https://github.com/mui-org/material-ui/pull/26666).

Reviewer 2: ```suggestion
- Remove the `span` element that wraps the children. Remove the `wrapper` classKey too. More details about [this change](https://github.com/mui-org/material-ui/pull/26666).
```

Reviewer 2: No objections to change the approach, one header by React component instead of UI component, but maybe the change should be done all at once, in a batch 🤷‍♂️
